src/runtime/virtcontainers/network_linux.go

```diff
@@ -252,6 +252,22 @@ func (n *LinuxNetwork) removeSingleEndpoint(ctx context.Context, s *Sandbox, idx
 	return nil
 }
 
+func (n *LinuxNetwork) endpointAlreadyAdded(netInfo *NetworkInfo) bool {
+	for _, ep := range n.eps {
+		// Existing endpoint
+		if ep.Name() == netInfo.Iface.Name {
+			return true
+		}
+		pair := ep.NetworkPair()
+		// Existing virtual endpoints
+		if pair != nil && (pair.TapInterface.Name == netInfo.Iface.Name || pair.TapInterface.TAPIface.Name == netInfo.Iface.Name) {
```

Comment:
Should compare `netInfo.Iface.Name` with `pair.TapInterface.Name`/`pair.TapInterface.TAPIface.Name`, or `netPair.VirtIface.Name`? 

Through https://github.com/kata-containers/kata-containers/blob/a55eb78c32e7461161b77b0663b2a737f4a83fba/src/runtime/virtcontainers/network.go#L354-L356 , the `netPair.VirtIface.Name`'s value will be set using the value of `netInfo.Iface.Name`.

Reply:
Let's match them all as I got `netInfo.Iface.Name == tap0_kata` here when testing.